src/NHibernate/Event/Default/DefaultInitializeCollectionEventListener.cs

```diff
@@ -62,9 +62,10 @@ public virtual void OnInitializeCollection(InitializeCollectionEvent @event)
 		}
 
 		/// <summary> Try to initialize a collection from the cache</summary>
-		private bool InitializeCollectionFromCache(object id, ICollectionPersister persister, IPersistentCollection collection, ISessionImplementor source)
+		private bool InitializeCollectionFromCache(
+			object collectionKey, ICollectionPersister persister, IPersistentCollection collection,
+			ISessionImplementor source)
```

Comment:
Renaming `id` to `collectionKey` for avoiding confusion. It may be something else than an identifier, it is a collection key, which can be a unique key property instead. The code calling this is actually supplying a key, not an id.